parcels/field.py

```diff
@@ -284,15 +285,20 @@ def from_netcdf(cls, filenames, variable, dimensions, indices=None, grid=None,
             else:
                 raise RuntimeError('interp_method is a dictionary but %s is not in it' % variable[0])
 
-        with NetcdfFileBuffer(lonlat_filename, dimensions, indices, netcdf_engine, field_chunksize=False, lock_file=False) as filebuffer:
+        if netcdf_engine == 'xarray':
+            FieldFileBuffer = XarrayFileBuffer
+        else:
+            FieldFileBuffer = NetcdfFileBuffer
+
+        with FieldFileBuffer(lonlat_filename, dimensions, indices, netcdf_engine) as filebuffer:
             lon, lat = filebuffer.read_lonlat
             indices = filebuffer.indices
             # Check if parcels_mesh has been explicitly set in file
             if 'parcels_mesh' in filebuffer.dataset.attrs:
                 mesh = filebuffer.dataset.attrs['parcels_mesh']
 
         if 'depth' in dimensions:
-            with NetcdfFileBuffer(depth_filename, dimensions, indices, netcdf_engine, interp_method=interp_method, field_chunksize=False, lock_file=False) as filebuffer:
+            with FieldFileBuffer(depth_filename, dimensions, indices, netcdf_engine, interp_method=interp_method) as filebuffer:
```

Comment:
this will be interesting how it's going to be solves. My suggestions here, explicit: Have 2 'field fielbuffers'! One that is used in the init-process to gather the file metadata - ideally a Netcdf buffer that is non-dask, just a regular-old netcdf-reader. The second 'field filebuffer' then would be for the actually data read access during field computations, and that one - if used in conjecture with Dask, needs to be file-locked (see Dask & xarray documentation on chunking).

Reply:
Good idea. I've renamed to `_grid_fb_class` in ff39e0e6fa6bc54ffe14de423bea09b48d556faf, as it mostly deals with reading the Grid (and could even go into grid.py??)